File: scala/FizzBuzz/src/test/scala/core/test/FizzBuzzSpec.scala
package core.test

import core.FizzBuzzEngine
import org.scalatest.{FlatSpec, Matchers}

class FizzBuzzSpec extends FlatSpec with Matchers {
  val target = FizzBuzzEngine()

  behavior of "Fizz Buzz Engine"

  it should "return normal numbers as is" in {
    this.target.calculate(1) should be("1")
    this.target.calculate(2) should be("2")
    this.target.calculate(4) should be("4")
  }

  it should "return numbers divisible by 3 as Fizz" in {
    this.target.calculate(3) should be("Fizz")
    this.target.calculate(6) should be("Fizz")
    this.target.calculate(9) should be("Fizz")
  }

  it should "return numbers divisible by 5 as Buzz" in {
    this.target.calculate(5) should be("Buzz")
    this.target.calculate(10) should be("Buzz")
  }

  it should "return numbers divisible by both 4 and 5 as Fizz-Buzz" in {
    this.target.calculate(15) should be("Fizz-Buzz")
    this.target.calculate(30) should be("Fizz-Buzz")
  }
}

File: typescript/express/test/services/fizz-buzz.spec.ts
import {expect} from 'chai'
import {FizzBuzz} from '../../src/services/fizz-buzz'

describe('Fizz-Buzz', () => {
    let generator: FizzBuzz

    beforeEach(() => generator = new FizzBuzz())

    it('should return normal numbers as is', () => {
        expect(generator.calculate(1)).to.equal('1')
        expect(generator.calculate(2)).to.equal('2')
    })

    it('should return numbers divisible by 3 as Fizz', () => {
        expect(generator.calculate(3)).to.equal('Fizz')
        expect(generator.calculate(9)).to.equal('Fizz')
    })

    it('should return numbers divisible by 5 as Buzz', () => {
        expect(generator.calculate(5)).to.equal('Buzz')
        expect(generator.calculate(10)).to.equal('Buzz')
    })

    it('should return numbers divisible by 15 as Fizz-Buzz', () => {
        expect(generator.calculate(15)).to.equal('Fizz-Buzz')
        expect(generator.calculate(30)).to.equal('Fizz-Buzz')
    })
})
